cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -248,15 +254,18 @@ impl NetworkRunnable for Cmd {
             } else {
                 let mut slop = vec![OsString::from(CONSTRUCTOR_FUNCTION_NAME)];
                 slop.extend_from_slice(&self.slop);
-                Some(
-                    arg_parsing::build_host_function_parameters(
-                        &stellar_strkey::Contract(contract_id.0),
-                        &slop,
-                        &entries,
-                        config,
-                    )?
-                    .2,
-                )
+                let params = arg_parsing::build_host_function_parameters(
+                    &stellar_strkey::Contract(contract_id.0),
+                    &slop,
+                    &entries,
+                    config,
+                )?;
+                match params {
+                    HostFunctionParameters::Params(p) => Some(p.2),
```

Comment:
is it possible the `p.2` will ever be `None`? I wonder if it's worth having a default arm in this match in case that happens.

Reply:
No, it shouldn't be (it will indicate contract's --help somehow passed into deploy)